### PR TITLE
refactor(0000-template.md): Update RFC template

### PR DESCRIPTION
**Pull request by jwerle**
This structure is more in line with the existing RFC documents. If this aligns with everyone else, we can merge this in, then migrate the existing RFC documents

### CODE REVIEW

File: 0000-template.md [hunk 1]
```diff
@@ -1,46 +1,99 @@
...
+<Topic Name>
+==============
 
-# Summary
+---
+Request For Comments: 0000
+---
+Author(s): <Author(s) First & Last>
+---
+Category: <Category>
```
Comment (cckelly) [inline 0000-template.md]: @jwerle Are categories arbitrary? As in will we have a predefined list of categories?

File: 0000-template.md [hunk 2]
```diff
@@ -1,46 +1,99 @@
...
+---
+Date: <Month>, <Year>
+---
 
-Brief explanation of the feature.
+## 0. Abstract
 
-# Basic example
+> An brief introductory explanation or abstract of the publication.
```
Comment (cckelly) [inline 0000-template.md]: Typo - should be 'a'.

File: 0000-template.md [hunk 3]
```diff
@@ -1,46 +1,99 @@
...
-this a breaking change?
+## 5. Real World Example
 
-# Unresolved questions
+> This section should present a real world example showing expected
+> outcomes described by the content of this document.
 
-Are there any lingering questions that need to be answered prior to a possible implementation of this RFC?
+> If an RFC describes a data structure, a human readable visual representation
+> should be show, such as a JSON strucuture.
```
Comment (cckelly) [inline 0000-template.md]: shown*

File: 0000-template.md (hunk 1 above)
Comment (jwerle) [reply 0000-template.md]: @cckelly yeah I imagine this to be somewhat like tags

File: 0000-template.md (hunk 2 above)
Comment (jwerle) [reply 0000-template.md]: :+1:

**Comment by jwerle** [comment]
@cckelly Updated!

**Pull request merged**